src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -175,6 +175,24 @@ internal bool EnableWebTransportAndH3Datagrams
         set => _enableWebTransportAndH3Datagrams = value;
     }
 
+    /// <summary>
+    /// Internal AppContext switch to toggle whether a request line can end with LF only instead of CR/LF.
+    /// </summary>
+    private bool? _disableHttp1LineFeedTerminators;
+    internal bool DisableHttp1LineFeedTerminators
+    {
+        get
+        {
+            if (!_disableHttp1LineFeedTerminators.HasValue)
+            {
+                _disableHttp1LineFeedTerminators = AppContext.TryGetSwitch("Microsoft.AspNetCore.Server.Kestrel.DisableHttp1LineFeedTerminators", out var disabled) && disabled;
```

Comment:
Nit: Let's make this an internal string or internal static method that can be referenced from the HttpParser ctor to make extra sure these strings never go out of sync.